R/dummy.R

```diff
@@ -284,11 +284,20 @@ bake.step_dummy <- function(object, new_data, ...) {
         na.action = na.pass
       )
 
-    indicators <-
-      model.matrix(
-        object = levels,
-        data = indicators
-      )
+    indicators <- tryCatch(
+      model.matrix(object = levels, data = indicators),
+      error = function(cnd) {
+        if (grepl("(vector memory|cannot allocate)", cnd$message)) {
```

Comment:
if the error message doesn't match the regex, it just flows through and gives the original error.